port/zos390/omrvmem.c

```diff
@@ -1352,9 +1363,18 @@ omrvmem_find_valid_page_size(struct OMRPortLibrary *portLibrary, uintptr_t mode,
 		uintptr_t defaultLargePageSize = 0;
 		uintptr_t defaultLargePageFlags = OMRPORT_VMEM_PAGE_FLAG_NOT_USED;
 
-		/* If the page type is NOT_USED (as with the -Xlp<size> options)
-		 * the legacy behavior is to default to FIXED pages.
-		 */
+		/* If the page type is PREFER_PAGEABLE try to get 1M pageable large pages. Otherwise, use a fixed large page. */
+		if ((IS_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE(validPageFlags)) && (ONE_M == validPageSize)) {
+
+			/* Try Pageable 1M large pages */
+			pageSizeFound = isLargePageSizeSupported(portLibrary, ONE_M, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
+			if (TRUE == pageSizeFound) {
+				SET_PAGE_TYPE(validPageFlags, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
+				goto _end;
+			}
+		}
+
+		/* If the page type is NOT_USED the legacy behavior is to default to FIXED pages. */
```

Comment:
`OMRPORT_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE` must be stripped out from `validPageFlags` (or even not set to it originally) at this point. `isLargePageSizeSupported()` uses exact match so it will fail always

Reply:
Right. That's been implemented.